### ISSUE TITLE
Potential false positive with SC2062

### ISSUE THREAD

**Issue by LiterallyUniqueLogin** [koalaman/shellcheck#1856]
#### For bugs
- Rule Id : SC2062
- My shellcheck version: 0.7.0 
- [X] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [X] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

![image](https://user-images.githubusercontent.com/1317013/75811355-e392bc00-5d51-11ea-84fd-cd6c3a88153a.png)

#### Here's what I wanted or expected to see:

I don't believe there should be any warning here (if I'm wrong, please teach me why =) ). The pattern I'm looking for is a dash, and it's already quoted. Maybe shellcheck is interpreting the quoted dash as a flag, and so thinks that the file list glob is a pattern when it's not?

This seems to [already have been noticed](https://github.com/koalaman/shellcheck/issues/517), but unsolved?

**Comment by gromgit** [comment]
> Maybe shellcheck is interpreting the quoted dash as a flag, and so thinks that the file list glob is a pattern when it's not?

That's exactly right. Your screenshot was ambiguous as to where the error was, so I ran it through shellcheck.net, which is a lot clearer:
```
grep '-' "$UKB"/microarray/*.sample > test.out
         ^-- SC2062: Quote the grep pattern so the shell won't interpret it.
```
This is why the `grep` man page says:

> `-e PATTERN, --regexp=PATTERN`
> Use *PATTERN* as the pattern.  If this option is used multiple times or is combined with the -f (--file) option, search for all patterns given.  **This option can be used to protect a pattern beginning with “-”.**

and the pedantically-correct way to write your code is:
```
grep -e - "$UKB"/microarray/*.sample > test.out
```

**Comment by LiterallyUniqueLogin** [comment]
Thanks for the quick response!

Sorry for the lack of clarity - the warning message was at the very bottom of the picture, I should have stated that.

And thanks for the clarification. I'd forgotten how the shell would interpret the quotes.

**Comment by gromgit** [comment]
> Sorry for the lack of clarity - the warning message was at the very bottom of the picture, I should have stated that.

Actually, the "lack of clarity" was about **what your editor's Shellcheck plugin was pointing to**. For those (like myself) unfamiliar with what your editor, there seem to be *two* possible locations: one at the end of `'-'`, and the other (correct) one at the beginning of your wildcarded path.

While screenshots are explicitly allowed, they're not favored precisely because of ambiguities like this. Plus, plain text is just easier for others to copy-paste and verify your issue.
